File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/longest-strictly-increasing-or-strictly-decreasing-subarray.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        maxInc, maxDec, length = 1, 1, 1
        for i in range(1, len(nums)):
            if nums[i] > nums[i - 1]:
                length += 1
            else:
                maxInc = max(maxInc, length)
                length = 1

        maxInc = max(maxInc, length)
        length = 1

        for i in range(1, len(nums)):
            if nums[i] < nums[i - 1]:
                length += 1
            else:
                maxDec = max(maxDec, length)
                length = 1
                
        maxDec = max(maxDec, length)
        return max(maxInc, maxDec)
        

File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/longest-strictly-increasing-or-strictly-decreasing-subarray.java
class Solution {
    public int longestMonotonicSubarray(int[] nums) {
        int maxInc = 1;
        int maxDec = 1;
        int len = 1;

        for (int i = 1; i < nums.length; i++) {
            if (nums[i] > nums[i - 1]) {
                len++;
            } else {
                maxInc = Math.max(maxInc, len); 
                len = 1; 
            }
        }
        maxInc = Math.max(maxInc, len); 

        len = 1; 


        for (int i = 1; i < nums.length; i++) {
            if (nums[i] < nums[i - 1]) {
                len++;
            } else {
                maxDec = Math.max(maxDec, len);
                len = 1; 
            }
        }
        maxDec = Math.max(maxDec, len);


        return Math.max(maxInc, maxDec);
    }
}
